### ISSUE TITLE
Rcpp exception with UTF-8 strings on Windows

### ISSUE THREAD

**Issue by qinwf** [qinwf/re2r#10]
This Rcpp issue will affect the error message for regular expression. 

```r
re2("this (is 测试")
#> Error: missing closing ): this (is 娴嬭瘯 
```

Here is an issue about related to this before. 

[[Rcpp-devel] Unicode on windows 1](http://lists.r-forge.r-project.org/pipermail/rcpp-devel/2013-August/006286.html)

 [[Rcpp-devel] Unicode on windows 2](http://lists.r-forge.r-project.org/pipermail/rcpp-devel/2013-August/006287.html)

The solution in the above mailing list posts can not solve the exception handling string problem.

I send an email to the Rcpp mailing list about this issue, and here is links to the discussion:

[[Rcpp-devel] Rcpp exception with UTF-8 strings on Windows 1](http://lists.r-forge.r-project.org/pipermail/rcpp-devel/2016-June/009291.html)

[[Rcpp-devel] Rcpp exception with UTF-8 strings on Windows 2](http://lists.r-forge.r-project.org/pipermail/rcpp-devel/2016-June/009292.html)

It seems that Rcpp will not fix this very soon. So I suggest to use the origin R-C API to rewrite existing codes.

**Comment by gagolews** [comment]
Just take a look at the way I handle UTF8 string input in `stringi`. It's pretty simple.
I suggest you `LinkingTo: stringi`, call `stri_enc_toutf8` on a given `SEXP` object and then play with `STRING_ELT` etc. on the resulting `SEXP`.

**Comment by qinwf** [comment]
Yes, I imported `stringi` and all of the input strings are processed by `stri_enc_toutf8`.

**Comment by qinwf** [comment]
I opened a [PR](https://github.com/RcppCore/Rcpp/pull/493) in Rcpp repo to make this fixable with a macro in Rcpp and it was merged.

**Comment by tdhock** [comment]
that's great that you contributed some code to Rcpp! Good job!

Now when you use the new macro this issue is fixed, right?

So now I would say we can keep the Rcpp interface, right? (we don't need to consider re-writing the re2r interface to use the standard Rinternals.h headers)

**Comment by gagolews** [comment]
Rcpp 0.12.6 is now on CRAN

http://dirk.eddelbuettel.com/blog/2016/07/19/#rcpp_0.12.6
